src/libs/actions/Card.ts

```diff
@@ -206,100 +206,88 @@ function revealVirtualCardDetails(cardID: number): Promise<ExpensifyCardDetails>
 
-    // TODO: uncomment this code when the API is ready
-    // const optimisticData: OnyxUpdate[] = [
-    //     {
-    //         onyxMethod: Onyx.METHOD.MERGE,
-    //         key: `${ONYXKEYS.COLLECTION.PRIVATE_EXPENSIFY_CARD_SETTINGS}${workspaceAccountID}`,
-    //         value: {
-    //             monthlySettlementDate: '',
-    //         },
-    //     },
-    // ];
-    //
-    // const successData: OnyxUpdate[] = [
-    //     {
-    //         onyxMethod: Onyx.METHOD.MERGE,
-    //         key: `${ONYXKEYS.COLLECTION.PRIVATE_EXPENSIFY_CARD_SETTINGS}${workspaceAccountID}`,
-    //         value: {
-    //             monthlySettlementDate: '',
-    //         },
-    //     },
-    // ];
-    //
-    // const failureData: OnyxUpdate[] = [
-    //     {
-    //         onyxMethod: Onyx.METHOD.MERGE,
-    //         key: `${ONYXKEYS.COLLECTION.PRIVATE_EXPENSIFY_CARD_SETTINGS}${workspaceAccountID}`,
-    //         value: {
-    //             monthlySettlementDate: null,
-    //         },
-    //     },
-    // ];
-    //
-    // const parameters = {
-    //     workspaceAccountID,
-    //     settlementFrequency: frequency,
-    // };
-    //
-    // API.write(WRITE_COMMANDS.UPDATE_CARD_SETTLEMENT_FREQUENCY, parameters, {optimisticData, successData, failureData});
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.PRIVATE_EXPENSIFY_CARD_SETTINGS}${workspaceAccountID}`,
+            value: {
+                monthlySettlementDate,
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.PRIVATE_EXPENSIFY_CARD_SETTINGS}${workspaceAccountID}`,
+            value: {
+                monthlySettlementDate,
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.PRIVATE_EXPENSIFY_CARD_SETTINGS}${workspaceAccountID}`,
+            value: {
+                monthlySettlementDate: currentFrequency,
+            },
+        },
+    ];
+
+    const parameters = {
+        workspaceAccountID,
+        settlementFrequency,
+    };
+
+    API.write(WRITE_COMMANDS.UPDATE_CARD_SETTLEMENT_FREQUENCY, parameters, {optimisticData, successData, failureData});
 }
 
-function updateSettlementAccount(workspaceAccountID: number, accountID: number) {
-    // TODO: remove this code when the API is ready
-    Onyx.merge(`${ONYXKEYS.COLLECTION.PRIVATE_EXPENSIFY_CARD_SETTINGS}${workspaceAccountID}`, {
-        paymentBankAccountID: accountID,
-    });
+function updateSettlementAccount(workspaceAccountID: number, domainName: string, settlementBankAccountID?: number, currentSettlementBankAccountID?: number) {
+    if (!settlementBankAccountID) {
+        return;
+    }
```

Comment:
If it's better to send some invalid value - like -1 and get the error from the BE, I'm ok with that I can change it. But in my other PRs I was requested not to do so, so I also applied it here. But since we don't have this kind of error handling anywhere through the app I think we should rely on the BE errors. 
cc @mountiny wdyt?

Reply:
Yeah I think its more common to do something and show error but I guess the problem with these is that its theoretical problem only, right? Like it should not happen that the bankAccountID is missing when we get this far?

Happy to change this to call with invalid value or the least we should do is to add some log